File: Math/CPP/convert_integer_to_the_sum_of_two_no-zero_integers.cpp
#include <vector>
using namespace std;

class Solution {
public:
    vector<int> getNoZeroIntegers(int n) {
        for (int i = 1; i < n; i++) {
            if (isNoZero(i) && isNoZero(n-i)) return vector<int>{i, n-i};
        }
        return vector<int>{};
    }
    bool isNoZero(int n) {
        while (n > 0) {
            if (!(n & 0x1)) {
                int nNext = n / 10;
                if (n - (10 * nNext) == 0) return false;
            }
            n /= 10;
        }
        return true;
    }
};

File: Math/Python/convert_integer_to_the_sum_of_two_no_zero_integers.py
class Solution:
    def getNoZeroIntegers(self, n: int) -> list[int]:
        for i in range(1, n+1):
            if self.check(i) and self.check(n-i):
                return [i, n-i]
        return []


    def check(self, n):
        while n:
            if not n & 1:
                nNext = n // 10
                if (n - (10 * nNext) == 0):
                    return False
                n = nNext
            n //= 10
        return True
